### ISSUE TITLE
Bug: adds lag when typing on remote SSH

### ISSUE THREAD

**Issue by aon** [DrMerfy/vscode-overtype#21]
Hey! So I noticed the extension adds lag when typing, both in overwrite or insert mode. When overwrite mode is enabled though, it doesn't work properly as the cursor doesn't move forward fast enough so it ends up writing in the same place several times, if that makes sense.

I can't really send a GIF over my work PC showing the behaviour so I'll send one when I'm on my personal one.

**Comment by cyraid** [comment]
Yeah I noticed this too. I do work on SSH all the time, and it's so bad that I had to uninstall it.. If only vscode had a native overtype.

**Comment by GitMensch** [comment]
Can't reproduce this, neither with small or big files. Is this still an issue or can we close it?

Note: I'm using [Open Remote - SSH](https://open-vsx.org/extension/jeanp413/open-remote-ssh) and this works like a charm.

How do you SSH exactly?
Can you please check if the CPU is crunching on the remote, for example with `top -u $LOGNAME` when this issue happens?

Also there is "report performance issue" available from the commands, when you see this issue please use that (performance issue, extensions, overtype, then possibly the current title and text with attachements, so we do have something to inspect / review.

**Comment by cyraid** [comment]
> Can't reproduce this, neither with small or big files. Is this still an issue or can we close it?
> 
> Note: I'm using [Open Remote - SSH](https://open-vsx.org/extension/jeanp413/open-remote-ssh) and this works like a charm.
> 
> How do you SSH exactly? Can you please check if the CPU is crunching on the remote, for example with `top -u $LOGNAME` when this issue happens?
> 
> Also there is "report performance issue" available from the commands, when you see this issue please use that (performance issue, extensions, overtype, then possibly the current title and text with attachements, so we do have something to inspect / review.

Back when I used it, haven't for a long while but, you gotta type really fast (I type quite fast when programming) you'll notice missed keys and such.

**Comment by GitMensch** [comment]
> Back when I used it, haven't for a long while but, you gotta type really fast (I type quite fast when programming) you'll notice missed keys and such.

Maybe this is a different issue - #23 (where it was reported that this may not be an issue any more)?

**Comment by cyraid** [comment]
> > Back when I used it, haven't for a long while but, you gotta type really fast (I type quite fast when programming) you'll notice missed keys and such.
> 
> Maybe this is a different issue - #23 (where it was reported that this may not be an issue any more)?

Oh maybe. Probably that other issue is what I had yeah.

**Comment by GitMensch** [comment]
Just a note: while I till have no clue how to "fix" anything here I now could reproduce it; switching the overtype mode did not changed much; disabling the extension led to normal speed in SSH editor.

> If only vscode had a native overtype.

https://github.com/microsoft/vscode/issues/1012 :-/

Actually we'd love to deprecate the extension... but fixes are still welcome, especially until this upstream issue is closed-fixed.
